src/Applications/LDAS_App/util/inputs/ASCAT_sm_mask/ascat_mask_maker.F90

```diff
@@ -0,0 +1,199 @@
+! This program reads in a NetCDF file containing ASCAT soil moisture masks available from:
+! Lindorfer, R., Wagner, W., Hahn, S., Kim, H., Vreugdenhil, M., Gruber, A., Fischer, M., & Trnka, M. (2023). 
+! Global Scale Maps of Subsurface Scattering Signals Impacting ASCAT Soil Moisture Retrievals (1.0.0) [Data set]. 
+! TU Wien. https://doi.org/10.48436/9a2y9-e5z14
+!
+! It provides the possibility to combine different masks (default case is combination of subsurface and wetland masks)
+! and interpolates onto a regular grid with a (hardwired) 0.1 degree lat/lon spacing and -90/-180 degree lower left 
+! corner used for quick indexing in the ASCAT observation reader QC routine.
+!
+! Author: AM Fox, March, 2024
+
+program ascat_mask_maker
+
+    use netcdf
+
+    implicit none
+
+    integer :: ncid, varid, dimid, ierr, len, N_gpi, dimids(2)
+    integer :: i, j, closest_index
+
+    integer(kind=1), dimension(:,:), allocatable :: mask_out
+    integer(kind=1)                              :: missing_value
+
+    real, dimension(:), allocatable :: asc_lon, asc_lat, cold_mask, wet_mask, veg_mask, subsurface_mask, combined_mask
```

Comment:
A brief look at the nc file suggests that the *_mask variables are stored as int8 in the original nc file.  Do we have to read them as real here and put them into the real array `combined_mask`?  The `where (wet_mask == 1)` condition below is risky for real numbers.  My sense is that everything should be in integer space if possible.

Reply:
Yes, all the mask variables should be integers. I'll change it.